src/components/CampaignInitialisation/CampaignInitialisationScreen.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useCallback,
+  useState
+} from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { StyleSheet, View } from "react-native";
+import { size } from "../../common/styles";
+import { NavigationProps } from "../../types";
+import { UpdateByRestartingContent } from "./UpdateByRestartingContent";
+import { UpdateFromAppStoreContent } from "./UpdateFromAppStoreContent";
+import { LoadingView } from "../Loading";
+import { useUpdateCampaignConfig } from "../../hooks/useUpdateCampaignConfig/useUpdateCampaignConfig";
+import { useAuthenticationContext } from "../../context/auth";
+import { useCheckVersion } from "../../hooks/useCheckVersion/useCheckVersion";
+import { useCheckUpdates } from "../../hooks/useCheckUpdates";
+
+const styles = StyleSheet.create({
+  wrapper: {
+    ...StyleSheet.absoluteFillObject,
+    justifyContent: "center",
+    alignItems: "center",
+    paddingHorizontal: size(5)
+  },
+  content: {
+    width: 512,
+    maxWidth: "100%",
+    marginTop: -size(4)
+  }
+});
+
+const RETRY_UPDATE_TIMES = 3;
+
+export const CampaignInitialisationScreen: FunctionComponent<NavigationProps> = ({
+  navigation
+}) => {
+  useEffect(() => {
+    Sentry.addBreadcrumb({
+      category: "navigation",
+      message: "CampaignInitialisationScreen"
+    });
+  }, []);
+
+  const { token, endpoint } = useAuthenticationContext();
+  const { fetchingState, updateCampaignConfig } = useUpdateCampaignConfig(
+    token,
+    endpoint
+  );
+  const checkVersion = useCheckVersion();
+  const checkUpdates = useCheckUpdates();
+
+  useEffect(() => {
+    updateCampaignConfig();
+    // updating the campaign config should only happen once when this screen is loaded
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I've also done this in other places of the code. I wonder what is the value of this rule in our codebase, since it's not always the best solution 🤔 Just a thought~

Reply:
Yea i've avoided this everywhere else, but I think in this case it's a legit one, since we actually really only want this to run once. We can keep this in mind as we develop!